File: python/pshexs/memcached.py

def test_output():
    return '{"memcached_test": "passed"}'


# from pshconfig import Config
# import pymemcache
#
#
# # Create a new Config object to ease reading the Platform.sh environment variables.
# # You can alternatively use os.environ yourself.
# config = Config()
#
# # Get the credentials to connect to the Memcached service.
# credentials = config.credentials('memcached')
#
# try:
#     # Try connecting to Memached server.
#     memcached = pymemcache.Client((credentials['host'], credentials['port']))
# #     $memcached = new Memcached();
# #     $memcached->addServer($credentials['host'], $credentials['port']);
#     memcached.set('Memcached::OPT_BINARY_PROTOCOL', True)
# #     $memcached->setOption(Memcached::OPT_BINARY_PROTOCOL, true);
#
#     key = "Deploy day"
#     value = "Friday"
#
#     # Set a value
#     memcached.set(key, value)
# #     $memcached->set($key, $value);
#
#     # Read it back.
#     test = memcached.get(key)
#
#     print('Found value {} for key {}'.format(test, key))
#
# except Exception as e:
#     print(e)

File: golang/examples/memcached.go
package examples

import (
  "fmt"
  psh "github.com/platformsh/config-reader-go/v2"
  "github.com/bradfitz/gomemcache/memcache"
)

func FormattedCredentialsMemcached(creds psh.Credential) (string, error) {
  formatted := fmt.Sprintf("%s:%d", creds.Host, creds.Port)
  return formatted, nil
}

func UsageExampleMemcached() string {

  // Create a NewRuntimeConfig object to ease reading the Platform.sh environment variables.
	// You can alternatively use os.Getenv() yourself.
  config, err := psh.NewRuntimeConfig()
  if err != nil {
    panic(err)
  }

  // Get the credentials to connect to the Solr service.
  credentials, err := config.Credentials("memcached")
  checkErr(err)

  // Retrieve formatted credentials for gomemcache.
  formatted, err := FormattedCredentialsMemcached(credentials)
  checkErr(err)

  // Connect to Memcached.
  mc := memcache.New(formatted)

  // Set a value.
  key := "Deploy_day"
  value := "Friday"

  err = mc.Set(&memcache.Item{Key: key, Value: []byte(value)})

  // Read it back.
  test, err := mc.Get("Deploy_day")

  return fmt.Sprintf("Found value <strong>%s</strong> for key <strong>%s</strong>.", test.Value, key)
}
